File: Hard/Median_of_two_sorted_arrays.py
#Max Hillyer 6/27/25
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        for i in nums2:
            nums1.append(i)
        nums1.sort()
        if len(nums1)%2 == 1:
            return nums1[len(nums1)//2]
        else:
            return((nums1[len(nums1)//2]  + nums1[len(nums1)//2 - 1]) /2)

File: Hard/median_of_two_sorted_arrays.c
// max hillyer 6/28/25

double findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size) {
    int totalSize = nums1Size + nums2Size;
    int merged[totalSize];
    int i = 0, j = 0, k = 0;
    while (i < nums1Size && j < nums2Size) {
        if (nums1[i] < nums2[j]) {
            merged[k++] = nums1[i++];
        } else {
            merged[k++] = nums2[j++];
        }
    }
    while (i < nums1Size) {
        merged[k++] = nums1[i++];
    }
    while (j < nums2Size) {
        merged[k++] = nums2[j++];
    }
    if (totalSize %2 == 0){
        return (merged[totalSize/2 - 1] + merged[totalSize/2]) / 2.0;
    } else {
        return merged[totalSize/2];
    }
}
